### PR TITLE
Search recipes page

### PR DESCRIPTION
**Pull request by FacuCristobalGlobant**
Implemented a new page that allows users to search for recipes by filtering with a set of inputs.

https://user-images.githubusercontent.com/109153713/207390780-7e8bde53-6539-44e8-893c-90ffa947df9d.mp4

Created tests for said features

image

### CODE REVIEW

File: lib/core/util/asset_constants.dart [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+  static const String nothingFont = 'NothingYouCouldDo';
   static const String timerIcon = 'assets/images/clock_icon.webp';
   static const String instagramIcon = 'assets/social/instagram.svg';
   static const String facebookIcon = 'assets/social/facebook.svg';
   static const String twitterIcon = 'assets/social/twitter.svg';
   static const String notFoundImage = 'assets/images/no_recipes_found.png';
   static const String likeIcon = 'assets/images/like_icon.png';
   static const String noImg = 'assets/images/no-img.webp';
+  static const String menuImage = 'assets/images/menu-image.jpg';
+  static const String searchBackground = 'assets/images/search_bg.jpg';
```
Comment (FranciscoGaraventa) [inline lib/core/util/asset_constants.dart]: Review image extension, maybe you can transform it to a webp

File: lib/data/repository/recipes_repository.dart [hunk 2]
```diff
@@ -31,6 +32,40 @@ class RecipesRepository implements IRepository {
...
+      final response = await apiService.getSearchResult(
+        parameters.cuisine,
+        parameters.mealType,
+        parameters.maxCarbs,
+        parameters.maxProtein,
+        parameters.maxCalories,
+        parameters.maxFat,
+        parameters.maxCaffeine,
+        parameters.maxCopper,
+        parameters.maxCalcium,
```
Comment (FranciscoGaraventa) [inline lib/data/repository/recipes_repository.dart]: Instead, to send all these parameters to throw the method, as an improvement you can create a Map<String, String> as a body (Maybe could call it "queryParameter") that is the name that you have inside apiService.

`queryParameters: {
        **ApiConstants.numberQueryParameter: ApiConstants.numberOfSearchResults**,
        'cuisine': cuisine,
        'mealType': mealType,
        'maxCarbs': maxCarbs,
        'maxProtein': maxProtein,
        'maxCalories': maxCalories,
        'maxFat': maxFat,
        'maxCaffeine': maxCaffeine,
        'maxCopper': maxCopper,
        'maxCalcium': maxCalcium,
      },`

File: lib/presentation/view/search.dart [hunk 3]
```diff
@@ -1,12 +1,459 @@
...
+                          ),
+                        )
+                        .toList(),
+                    onChanged: (value) {
+                      filterController.updateMealType(
+                        MealTypeEnum.values.firstWhere(
+                          (e) => e.propertyName == value,
+                        ),
+                      );
+                    },
```
Comment (FranciscoGaraventa) [inline lib/presentation/view/search.dart]: You can create a new widget and as parameter you can use value` (filterController.mealType.value.propertyName or filterController.cuisine.value.PropertyName)`
item list could be (MealTypeEnum and CuisineEnum) and also you can send onChanged function too

File: lib/presentation/view/search.dart [hunk 4]
```diff
@@ -1,12 +1,459 @@
...
+                    max: SlidersEnum.maxCarbs.maxValue,
+                    min: SlidersEnum.maxCarbs.minValue,
+                    value: filterController.maxCarbs.value,
+                    onChanged: (value) {
+                      filterController.updateMaxCarbs(
+                        value.floorToDouble(),
+                      );
+                    },
+                  ),
+                ),
```
Comment (FranciscoGaraventa) [inline lib/presentation/view/search.dart]: You can create a new widget, using title, trailing, max and min values for slider and also onChanged method call

File: lib/presentation/view/search.dart [hunk 5]
```diff
@@ -1,12 +1,459 @@
...
+                Padding(
+                  padding: Dimensions.smallHorizontalPadding,
+                  child: RecipeGrid(
+                    listOfRecipes: state!,
+                    doOnTap: () {
+                      {
+                        searchResultsController.resetSearchResults();
+                        filterController.resetFilters();
+                      }
+                    },
```
Comment (FranciscoGaraventa) [inline lib/presentation/view/search.dart]: I think here you have extras {}

File: lib/presentation/widget/recipe_grid_widget.dart [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    listOfRecipes.forEach((recipe) {
+      children.add(
+        Container(
+          padding: Dimensions.smallerPadding,
+          width: similarRecipeWidth,
+          child: GestureDetector(
+            onTap: () {
+              if (doOnTap != null) {
+                doOnTap!();
+              }
```
Comment (FranciscoGaraventa) [inline lib/presentation/widget/recipe_grid_widget.dart]: Here instead to check if the function is not null, you can use doOnTap?.call(), this is going to make the call if the is not null and if the is null omits this line

**Review by CIFUM007 — APPROVED** [review]

**Review by MarcosCGlobant — APPROVED** [review]

File: lib/core/util/asset_constants.dart (hunk 1 above)
Comment (FacuCristobalGlobant) [reply lib/core/util/asset_constants.dart]: Images converted and compressed.

File: lib/data/repository/recipes_repository.dart (hunk 2 above)
Comment (FacuCristobalGlobant) [reply lib/data/repository/recipes_repository.dart]: Done, a Map is created in the repository and passed to the apiService as a parameter.

File: lib/presentation/view/search.dart (hunk 3 above)
Comment (FacuCristobalGlobant) [reply lib/presentation/view/search.dart]: Created a DropDownFilter widget.

File: lib/presentation/view/search.dart (hunk 4 above)
Comment (FacuCristobalGlobant) [reply lib/presentation/view/search.dart]: Created SliderFilter widget.

File: lib/presentation/view/search.dart (hunk 5 above)
Comment (FacuCristobalGlobant) [reply lib/presentation/view/search.dart]: I did, I removed them.

File: lib/presentation/widget/recipe_grid_widget.dart (hunk 6 above)
Comment (FacuCristobalGlobant) [reply lib/presentation/widget/recipe_grid_widget.dart]: Chnaged

**Review by FranciscoGaraventa — APPROVED** [review]
LG

**Pull request merged**